### ISSUE TITLE
More fine-grained documentation needed for `torchao.autoquant()`

### ISSUE THREAD

**Issue by suvadityamuk** [pytorch/ao#907]
There is an important need for better documentation of options, features, and args for `torchao.autoquant()` along with some important gotchas, i.e. around the `error_on_unseen` flag among others.

**Comment by jerryzh168** [comment]
cc @HDCharles can you take a look

maybe adding `error_on_unseen` to the official list of args: https://github.com/pytorch/ao/blob/53b6b787bb035f1abc3c4804dd22c09ee41e7a0d/torchao/quantization/autoquant.py#L558 would be helpful

**Comment by jerryzh168** [comment]
@suvadityamuk please let us know if you'd like to work on this as well, also it would be helpful if you can provide a bit more detail on what the existing doc is missing as well, apart from `error_on_unseen`

**Comment by suvadityamuk** [comment]
> @suvadityamuk please let us know if you'd like to work on this as well, also it would be helpful if you can provide a bit more detail on what the existing doc is missing as well, apart from `error_on_unseen`

Yeah sure, happy to work on this. I'll try to go and check again further about what is missing as well. 

cc: @msaroufim We spoke today at the PyTorch Conference about this. Let me know any next steps about this as well, thanks!
